examples/node-groups/self-managed-node-groups-spot/main.tf

```diff
@@ -0,0 +1,186 @@
+  tenant          = var.tenant      # AWS account name or unique id for tenant
+  environment     = var.environment # Environment area eg., preprod or prod
+  zone            = var.zone        # Environment with in one sub_tenant or business unit
+  cluster_version = var.cluster_version
+
+  vpc_cidr     = "10.0.0.0/16"
+  vpc_name     = join("-", [local.tenant, local.environment, local.zone, "vpc"])
+  azs          = slice(data.aws_availability_zones.available.names, 0, 3)
+  cluster_name = join("-", [local.tenant, local.environment, local.zone, "eks"])
+
+  terraform_version = "Terraform v1.0.1"
+
+  ca_expander          = "priority"
+  ca_priority_expander = <<-EOT
+    100:
+      - .*-spot-.*
+    10:
+      - .*
+  EOT
+}
+
+module "aws_vpc" {
+  source  = "terraform-aws-modules/vpc/aws"
+  version = "v3.2.0"
+
+  name = local.vpc_name
+  cidr = local.vpc_cidr
+  azs  = local.azs
+
+  public_subnets  = [for k, v in local.azs : cidrsubnet(local.vpc_cidr, 8, k)]
+  private_subnets = [for k, v in local.azs : cidrsubnet(local.vpc_cidr, 8, k + 10)]
+
+  enable_nat_gateway   = true
+  create_igw           = true
+  enable_dns_hostnames = true
+  single_nat_gateway   = true
+
+  public_subnet_tags = {
+    "kubernetes.io/cluster/${local.cluster_name}" = "shared"
+    "kubernetes.io/role/elb"                      = "1"
+  }
+
+  private_subnet_tags = {
+    "kubernetes.io/cluster/${local.cluster_name}" = "shared"
+    "kubernetes.io/role/internal-elb"             = "1"
+  }
+}
+#---------------------------------------------------------------
+# Example to consume eks_blueprints module
+#---------------------------------------------------------------
+module "eks_blueprints" {
+  source = "../../.."
+
+  tenant            = local.tenant
+  environment       = local.environment
+  zone              = local.zone
+  terraform_version = local.terraform_version
+
+  # EKS Cluster VPC and Subnet mandatory config
+  vpc_id             = module.aws_vpc.vpc_id
+  private_subnet_ids = module.aws_vpc.private_subnets
+
+  # EKS CONTROL PLANE VARIABLES
+  cluster_version = local.cluster_version
+
+  # EKS SELF MANAGED NODE GROUPS
+  self_managed_node_groups = {
+    ondemand_2vcpu_8mem = {
+      node_group_name    = "smng-ondemand-2vcpu-8mem"
+      instance_types     = ["m5.large"]
+      min_size           = "2"
+      subnet_ids         = module.aws_vpc.private_subnets
+      launch_template_os = "amazonlinux2eks" # amazonlinux2eks or bottlerocket
+    }
+
+    spot_2vcpu_8mem = {
+      node_group_name    = "smng-spot-2vcpu-8mem"
+      capacity_type      = "spot"
+      capacity_rebalance = true
+      instance_types     = ["m5.large", "m4.large", "m6a.large", "m5a.large", "m5d.large"]
+      min_size           = "0"
+      subnet_ids         = module.aws_vpc.private_subnets
+      launch_template_os = "amazonlinux2eks" # amazonlinux2eks or bottlerocket
+      k8s_taints         = [{ key = "spotInstance", value = "true", effect = "NO_SCHEDULE" }]
```

Comment:
Does this work? `k8s_taints` is not a valid input parameter for self-managed node groups.

Currently taints can be applied through `kubelet_extra_args` . I will update the docs for this

Reply:
@vara-bonthu yes, it works, but if the preference it's doing it through `kubelet_extra_args` I'll do it.